packages/trader/src/AppV2/Components/PurchaseButton/purchase-button.tsx

```diff
@@ -58,14 +59,17 @@ const PurchaseButton = observer(() => {
     const purchase_button_content_props = {
         currency,
         has_open_accu_contract,
-        is_accumulator,
-        is_high_low,
         is_multiplier,
-        is_touch,
         is_turbos,
-        is_vanilla_fx,
         is_vanilla,
     };
+    const has_no_button_content =
+        is_vanilla ||
+        is_vanilla_fx ||
+        is_turbos ||
+        is_high_low ||
```

Comment:
should this be wrapped in useMemo

 const has_no_button_content = useMemo(() => {
        return (
            is_vanilla ||
            is_vanilla_fx ||
            is_turbos ||
            is_high_low ||
            is_touch ||
            (is_accumulator && !has_open_accu_contract)
        );
    }, [is_vanilla, is_vanilla_fx, is_turbos, is_high_low, is_touch, is_accumulator, has_open_accu_contract]);

Reply:
I think no, it shouldn't.
1. Because child (`PurchaseButtonContent`) is not wrapped with `React.memo`, so it won't stop rerenders + even if it would be wrapped, then it will make sense to wrap non-primitives. 
2. Also, `PurchaseButtonContent` is rendering text content of the button, there is noting criminal in rerendering
3. If everything will be wrapped with memos and callbacks, it wouldn't help, because we are passing object info to`PurchaseButtonContent`. Inside it we are using obj_contract_basis.payout (type string) fields which is changing every second. In case of Vanillas, it contains PPP, so 19.260765 ->1s-->19.240708 -> 1s -> new value
4. I wrapped has_no_button_content with useMemo and log rerenders inside of child. It's still rerendering

Reply:
okay yeah got it, has_no_button_content is  primitives value, so does not matter